config/src/test/kotlin/org/springframework/security/config/web/servlet/SecurityContextDslTests.kt

```diff
@@ -0,0 +1,182 @@
+        override fun configure(http: HttpSecurity) {
+            // @formatter:off
+            http {
+                securityContext { }
+            }
+            // @formatter:on
+        }
+
+        @Bean
+        open fun objectPostProcessor(): ObjectPostProcessor<Any> = objectPostProcessor
+
+        companion object {
+            var objectPostProcessor: ObjectPostProcessor<Any> = spyk(ReflectingObjectPostProcessor())
+
+            class ReflectingObjectPostProcessor : ObjectPostProcessor<Any> {
+                override fun <O> postProcess(`object`: O): O = `object`
+            }
+        }
+    }
+
+    @Test
+    fun `security context when invoked twice then uses original security context repository`() {
+        spring.register(DuplicateDoesNotOverrideConfig::class.java).autowire()
+        every { DuplicateDoesNotOverrideConfig.SECURITY_CONTEXT_REPOSITORY.loadContext(any<HttpRequestResponseHolder>()) } returns mockk<SecurityContext>(relaxed = true)
+        mvc.perform(get("/"))
+        verify(exactly = 1) { DuplicateDoesNotOverrideConfig.SECURITY_CONTEXT_REPOSITORY.loadContext(any<HttpRequestResponseHolder>()) }
+    }
+
+
+    @EnableWebSecurity
+    open class DuplicateDoesNotOverrideConfig : WebSecurityConfigurerAdapter() {
+        override fun configure(http: HttpSecurity) {
+            // @formatter:off
+            http {
+                securityContext {
+                    securityContextRepository = SECURITY_CONTEXT_REPOSITORY
+                }
+                securityContext { }
+            }
+            // @formatter:on
+        }
+
+        companion object {
+            var SECURITY_CONTEXT_REPOSITORY = mockk<SecurityContextRepository>(relaxed = true)
+        }
+    }
+
+    @Test
+    fun `security context when security context repository not configured then does not throw exception`() {
+        spring.register(SecurityContextRepositoryDefaultsSecurityContextRepositoryConfig::class.java).autowire()
+        assertDoesNotThrow { mvc.perform(get("/")) }
+    }
+
+    @EnableWebSecurity
+    open class SecurityContextRepositoryDefaultsSecurityContextRepositoryConfig : WebSecurityConfigurerAdapter(true) {
+        override fun configure(http: HttpSecurity) {
+            // @formatter:off
+            http {
+                addFilterAt<WebAsyncManagerIntegrationFilter>(WebAsyncManagerIntegrationFilter())
+                anonymous { }
+                securityContext { }
+                authorizeRequests {
+                    authorize(anyRequest, permitAll)
+                }
+                httpBasic { }
+            }
+            // @formatter:on
+        }
+
+        override fun configure(auth: AuthenticationManagerBuilder) {
+            // @formatter:off
+            auth
+                    .inMemoryAuthentication()
+                    .withUser("user").password("password").roles("USER")
+            // @formatter:on
+        }
+    }
+
+    @Test
+    fun requireExplicitSave() {
```

Comment:
This test name doesn't follow the expected format. The test name should describe the configuration and the expected result. The other tests in this file are named correctly.